src/net_processing.cpp

```diff
@@ -4260,6 +4498,21 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         const uint256& hash = peer->m_wtxid_relay ? wtxid : txid;
         AddKnownTx(*peer, hash);
 
+        if (auto num_broadcasted = m_tx_for_private_broadcast.Remove(ptx)) {
```

Comment:
Just noting here that this is also racy (like reattempt). If we have more than 1 connection open and more than 1 succeeds, it will decrement the extra successful connections count.

Reply:
Yes, this loose handling makes it possible to do the job with less code. My reasoning is that the worst outcome from the race is acceptable.

I am looking into keeping track of more stuff with respect to the future RPC stats (see at the bottom of https://github.com/bitcoin/bitcoin/pull/29415#discussion_r2042286270) which would also give us more fine grained control here. If it looks reasonable and is not too much code, I will propose the change in https://github.com/bitcoin/bitcoin/pull/29415#discussion_r2042286270.